File: src/main/java/com/example/midterm_firstname_lastname/HistoryActivity.java
package com.example.midterm_firstname_lastname;

import android.content.Context;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.example.myapplication.R;

import java.util.ArrayList;
import java.util.Collections;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

public class HistoryActivity extends AppCompatActivity {

    private ListView historyList;
    private ArrayAdapter<String> adapter;
    private final List<String> items = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_history);

        historyList = findViewById(R.id.historyList);
        adapter = new ArrayAdapter<>(this, android.R.layout.simple_list_item_1, items);
        historyList.setAdapter(adapter);

        loadHistory();
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.main_menu, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == R.id.action_clear_all) {
            clearHistory();
            return true;
        }
        return super.onOptionsItemSelected(item);
    }

    private void loadHistory() {
        items.clear();
        Set<String> saved = getSharedPreferences("table_history", Context.MODE_PRIVATE)
                .getStringSet("numbers", new HashSet<>());
        List<Integer> nums = new ArrayList<>();
        for (String s : saved) {
            try { nums.add(Integer.parseInt(s)); } catch (Exception ignored) {}
        }
        Collections.sort(nums, Collections.reverseOrder());
        for (Integer n : nums) {
            items.add("Table of " + n);
        }
        adapter.notifyDataSetChanged();
    }

    private void clearHistory() {
        getSharedPreferences("table_history", Context.MODE_PRIVATE).edit().clear().apply();
        loadHistory();
        Toast.makeText(this, "History cleared", Toast.LENGTH_SHORT).show();
    }
}

File: src/main/java/com/example/myapplication/HistoryActivity.kt
package com.example.myapplication

import android.content.Context
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.widget.ArrayAdapter
import android.widget.ListView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class HistoryActivity : AppCompatActivity() {

    private lateinit var historyListView: ListView
    private lateinit var adapter: ArrayAdapter<String>
    private val items: MutableList<String> = mutableListOf()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_history)

        historyListView = findViewById(R.id.historyList)
        adapter = ArrayAdapter(this, android.R.layout.simple_list_item_1, items)
        historyListView.adapter = adapter

        loadHistory()
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.main_menu, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            R.id.action_clear_all -> {
                clearHistory()
                true
            }
            else -> super.onOptionsItemSelected(item)
        }
    }

    private fun loadHistory() {
        items.clear()
        val prefs = getSharedPreferences("table_history", Context.MODE_PRIVATE)
        val saved = prefs.getStringSet("numbers", emptySet()) ?: emptySet()
        // Show in reverse (latest first)
        items.addAll(saved.sortedByDescending { it.toIntOrNull() ?: 0 }.map { "Table of $it" })
        adapter.notifyDataSetChanged()
    }

    private fun clearHistory() {
        val prefs = getSharedPreferences("table_history", Context.MODE_PRIVATE)
        prefs.edit().clear().apply()
        loadHistory()
        Toast.makeText(this, "History cleared", Toast.LENGTH_SHORT).show()
    }
}


